### ISSUE TITLE
SQL-Sanitize Fails in Drush 9.0.0-beta4 

### ISSUE THREAD

**Issue by mikemadison13** [drush-ops/drush#3013]
After upgrading to 9.0.0-beta4, we are attempting to sanitize databases during syncs.

it would appear that the --sanitize flag was removed in this commit (https://github.com/drush-ops/drush/commit/19543e5b5ebecc68f71b782fa3f57d954af4632f#diff-fb2c277772bf5979ffd2cd9676de3c88). Note that it is still listed out here as a flag -- https://drushcommands.com/drush-9x/core/sql-sync/

As far as  we can tell (see https://github.com/drush-ops/drush/blob/master/src/Drupal/Commands/sql/SanitizeCommands.php) sql-sanitize is still supposed to be a command that functions, however when doing a drush list it doesn't appear in the sql commands, and it doesn't work when running it.

```
 sql:                                                                                                       
   sql-connect                  A string for connecting to the DB.                                          
   sql-create                   Create a database.                                                          
   sql-drop                     Drop all tables in a given database.                                        
   sql-cli (sqlc)               Open a SQL command-line interface using Drupal's credentials.               
   sql-query (sqlq)             Execute a query against a database.                                         
   sql-dump                     Exports the Drupal DB as SQL using mysqldump or equivalent.                 
   sql-sync                     Copy DB data from a source site to a target site. Transfers data via rsync. 
```

**Comment by weitzman** [comment]
Correct, that flag was removed. You should sql-sync and then sql-sanitize.
You now need a functional Drupal site in order to sanitize. I think thats
why the command is not being listed. This was impossible to avoid.

**Comment by greg-1-anderson** [comment]
The new architecture for Drush 9 instantiates Drupal-specific Drush commands via the Drupal services.yml mechanism, from information contained in a drush.services.yml file. This allows Drush commands to participate in Drupal dependency injection.

Because Drupal-specific commands now expect Drupal objects to be injected into them, Drush does not instantiate any of these types of commands unless there is a valid site available.

It is an aspirational goal to instead instantiate a proxy command for all of the Drupal-specific commands, so that they are not completely invisible when inspecting Drush from a non-bootstrapped environment. Running such a proxy command would only print a message advising the user how to select a site, or make their selected site bootstrap-able.
